packages/scope-manager/src/scope/ScopeBase.ts

```diff
@@ -88,7 +88,7 @@ function isStrictScope(
       break;
     }
 
-    if (stmt.directive === 'use strict') {
+    if ('directive' in stmt && stmt.directive === 'use strict') {
```

Comment:
It seems that both doing

```ts
if (stmt?.directive === 'use strict') {
```
or
```ts
if ('directive' in stmt && stmt.directive === 'use strict') {
```
doesn't work, although the latter should work according to this [TypeScript playground] 🤔(https://www.typescriptlang.org/play?ssl=14&ssc=64&pln=14&pc=5#code/KYOwrgtgBAggygFQPoDkDyARAokhBNABSzigG8AoKKqLADwAcAnYAZxYEsB7EOAFwENewCKF5QAvFADkdJqw7c+g4aKkAacgF9yvAJ71gUDO2YBjXuwBuhyRWpQ9BgFyxEqTDnxE4AOlnM2Lh4BIREQXgBuSmoAExNgcytgFxZeRnYQAHMo7UdDf3kgpVDRCTJoqjyXeGR0bFxCYj8GAIVg5TDIrR19Q2KVcLLjMwtrKAAfGhbCxRCBrvJTblSoUwALBIBrfs6ygApUjtEXHdEASnEAPjtqdgAzPak4kaSpKAyoQ5LBgDIfz7mnR8zwSoxs4kkUikZwoFSg2k0QA)

Any help here is welcome.